File: dataBase（数据库）/0SaveUploadFile/InsertionSort3.java


public class InsertionSort {
    public static void insertionSort(int[] nums) {
        int n = nums.length;
        int j;
        for (int i = 1; i < n; i++) {
            int temp = nums[i];
            for (j = i - 1; j >= 0 && temp < nums[j]; j--) {
                nums[j + 1] = nums[j];
            }
            nums[j + 1] = temp;
        }
    }
}

File: dataBase（数据库）/0SaveUploadFile/insertionsort3.py
#辅助函数——交换两个数组
def exchange(a,i,j):
    temp = a[i]
    a[i] = a[j]
    a[j] = temp

#插入排序函数
def insertion(a):
    length = len(a)
    for i in range(1,length):
        j = i
        while (j>0 and a[j]<a[j-1]):
            exchange(a,j,j-1)
            j -= 1

